### DISCUSSION TITLE
Adding data option to webhook / flows

### DISCUSSION THREAD

**Discussion by laurnts** [directus/directus#19526] [Feature Requests]
### Summary

Sometimes when making an API call, we don't necessarily to always include a body. For instance in order to trigger a Rundeck API call, Rundeck requires their options to be send through data instead of body.
<img width="336" alt="Screenshot 2023-08-25 at 13 09 17" src="https://github.com/directus/directus/assets/5869140/2beef17b-b045-4154-87a2-21032428d688">

### Motivation

expandibility of the flow - webhook of Directus towards making integrations with other tools

### Detailed Design

<img width="833" alt="Screenshot 2023-08-25 at 13 06 52" src="https://github.com/directus/directus/assets/5869140/135044d7-1674-4cac-9bf3-b4994ca60f93">
maybe something like this to add data fields (similar to body but for data)

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

I dont know

### Alternatives

I dont know

### Adoption Strategy

I dont know

**Comment by DanielBiegler** [comment]
> ### Alternatives
> I dont know

You could solve this by making your own custom operation: https://docs.directus.io/extensions/operations.html

In there you can do anything you'd like.

On adding new features we follow a 20/80 rule of thumb where we consider adding features that benefit roughly 80% of users. I'm not sure that many people have this specific use case? Let's see if theres interest! :)

In the meantime you can solve this with an extension. Cheers

**Comment by br41nslug** [comment]
Not quite sure if i understand this feature request 🤔 the `--data-raw` or `--data` flags on curl are not different from the `body` and are used to put data in the `body` of an http request. The screenshot shows a regular json body the same as what the Directus API does. 
```bash
curl --location 'http://localhost:8055/items/test' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer adminToken' \
--data-raw '{"some":"JSON data"}'
```

**Reply by laurnts** [reply]
@br41nslug your comments actually enlighten the situation and I finally managed to get it to work.
Finally I got the 200 ok. So this is how I fixed it
<img width="824" alt="Screenshot 2023-08-25 at 14 45 00" src="https://github.com/directus/directus/assets/5869140/904189bf-1456-4e65-8c96-276392b2c4ba">

This is the options
<img width="821" alt="Screenshot 2023-08-25 at 14 46 29" src="https://github.com/directus/directus/assets/5869140/603f36f1-767f-4a21-9580-6c23b560dc67">

This is the payload
<img width="836" alt="Screenshot 2023-08-25 at 14 47 27" src="https://github.com/directus/directus/assets/5869140/ec4b10fc-5d2d-4e3f-98b9-fdd130937bcc">

**Reply by laurnts** [reply]
This feature request can be closed

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
